frontend/public/components/resource-quota.jsx

```diff
@@ -30,11 +124,77 @@ const Row = ({obj: rq}) => <div className="row co-resource-list__item">
   </div>
 </div>;
 
+const CapacityIcon = ({capacity}) => {
+  let capacityIconClass = 'pficon-unknown';
+  if (capacity === 0) {
+    capacityIconClass = 'fa-circle-thin co-m-resource-quota-empty';
+  } else if (capacity > 0 && capacity < .5) {
+    capacityIconClass = 'pficon-resources-almost-empty';
+  } else if (capacity >= .5 && capacity < 1){
+    capacityIconClass = 'pficon-resources-almost-full';
+  } else if (capacity === 1) {
+    capacityIconClass = 'pficon-resources-full';
+  } else if (capacity > 1) {
+    capacityIconClass = 'pficon-warning-triangle-o';
+  }
+  const classPrefix = capacityIconClass.startsWith('pficon') ? 'pficon' : 'fa';
+  return <i className={`${classPrefix} ${capacityIconClass}`} aria-hidden="true" />;
+};
+
+const getCapacity = (used, max, resourceType) => {
+  let capacity = 0;
+  if (max && used) {
+    let unitType = units.getUnitType(max);
+    if (!unitType) {
+      unitType = units.getUnitType(used);
+    }
+    // max cpu values do not have units, convert number and add unit for capacity calculation
```

Comment:
This is wrong. Max CPU values CAN have a unit that we need to account for. We need to make sure we're using the utilties from units for any compute resource.

Reply:
If there is a unit for max cpu it will be used as is, this only handles when there is not a unit.

Reply:
But min could also not have a unit. There's nothing special about max CPU. It's also possible to specify memory values in bytes with no units.

Reply:
I am seeing examples where min has a unit, but max does not:

![image](https://user-images.githubusercontent.com/12733153/48725048-f8008300-ebf8-11e8-9a42-da4f491cc17e.png)
![image](https://user-images.githubusercontent.com/12733153/48725103-21211380-ebf9-11e8-9e49-4b486c43ce5d.png)

Reply:
Right, but that's just one specific example. It's not required for any CPU or memory value. `requests.cpu` could be "1" and `limits.cpu` could be 1500m.

You should just be able to always call `dehumanize` and work with the number that comes back.

Reply:
> Well, the `200m` will result in a unitType of 'numeric', but calling `units.dehumanize('1', 'numeric')` results in a 0 -the logic in dehumanize doesn't handle this case.

You need to pass the CPU type for it to handle it correctly. It's not numeric. It's CPU.

I don't think you can infer the unit type. You need to specifically pass CPU for CPU, memory for memory, etc.

Reply:
I do not see a CPU type in  https://github.com/openshift/console/blob/master/frontend/public/components/utils/units.js#L6  ?

Reply:
The two CPU dehumanize examples I see are using 'numeric': https://github.com/openshift/console/blob/master/frontend/public/components/pod.jsx#L161
https://github.com/openshift/console/blob/master/frontend/public/components/pod.jsx#L165

Reply:
> The two CPU dehumanize examples I see are using 'numeric': 

OK, I see now. I think that's just a bug. We need to add correct handling for CPU units.

Reply:
> OK, I see now. I think that's just a bug. We need to add correct handling for CPU units.

Ok, added `dehumanizeCPU` method